util/include/util/dbutil.h

```diff
@@ -81,6 +81,13 @@ std::string connStrComponent(
   const std::string& connStr_,
   const std::string& key_);
 
+/**
+ * This function returns the database driver name being used. Currently "pgsql"
+ * or "sqlite" can be returned depending on the build option giving which
+ * database system to use.
+ */
+std::string getDbDriver();
```

Comment:
Please, tell me, if I'm wrong, but I don't like to believe that `inline` keyword before a function definition has anything to do with the optimization methodology named "inlining". I don't know why this name was chosen for this keyword. Every blog and description about this keyword states that this is a hint for the compiler, but does not oblige it to do inlining. If this keyword (1) can't force the compiler do the inlining and (2) inlining optimization is done by the compiler in some cases even if `inline` keyword is missing, these are two statements which prove to me that "inlining" and `inline` keywords are independent. The only observable difference what this keyword makes is its real purpose: making the function definition local to the translation unit:
```.cpp
f.h
---
#ifndef F_H
#define F_H
/* inline */ void f() {}  // Note that this is a definition.
#endif

a.cpp
-----
#include "f.h"
int main() { f(); }

b.cpp
-----
#include "f.h"
void foobar() { f(); }

compilation
-----------
g++ a.cpp b.cpp  # Linker error if function f() is not inline.
```
It would be worth checking a compiler if this keyword plays any role during optimization, but I've never seen an evidence of this connections. This is a keyword for the linker. By the way this linkage rule is applied to the class methods when the standard says that class methods are automatically considered as if `inline` was provided. That's why they link even if their definition is written in the header file for some reason.

Reply:
It is in the standard, as for [C++11](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2011/n3242.pdf):
> § 7.1.2.2 A function declaration (8.3.5, 9.3, 11.3) with an **inline** specifier declares an inline function. The inline specifier indicates to the implementation that inline substitution of the function body at the point of call is to be preferred to the usual function call mechanism. An implementation is not required to perform this inline substitution at the point of call; however, even if this inline substitution is omitted, the other rules for inline functions defined by 7.1.2 shall still be respected.

While as you observed, inlining is not mandatory for the compiler, **inlining is hinted by the inline specifier.**

Reply:
I've been experimenting a bit with this issue. The results were as diverse as expected :) So yes, sometimes `inline` provokes inlining, unless -O0 prevents it. Sometimes it does the inlining and also leaves the function definition in the binary code as a weak symbol which explains how it solves the linker issue. Sometimes it doesn't provoke inlining if the function is too complex. So the optimizer would optimize the code in our specific case, but I added `inline` keyword in case some other optimization can be done here we don't know about.